proc/decom.go

```diff
@@ -3,20 +3,21 @@ package proc
 import (
 	"fmt"
 	"github.com/AarC10/GSW-V2/lib/ipc"
+	"github.com/AarC10/GSW-V2/lib/tlm"
 	"net"
 	"strconv"
 )
 
-func getIpcShmHandler(packet TelemetryPacket, write bool) (*ipc.IpcShmHandler, error) {
+func getIpcShmHandler(packet tlm.TelemetryPacket, write bool) (*ipc.IpcShmHandler, error) {
 	handler, err := ipc.CreateIpcShmHandler(strconv.Itoa(packet.Port), GetPacketSize(packet), write)
 	if err != nil {
-		return nil, fmt.Errorf("Error creating shared memory handler: %v", err)
+		return nil, fmt.Errorf("error creating shared memory handler: %v", err)
 	}
 
 	return handler, nil
 }
 
-func TelemetryPacketWriter(packet TelemetryPacket) {
+func TelemetryPacketWriter(packet tlm.TelemetryPacket, outChannel chan []byte) {
```

Comment:
again, some of the errors that can happen are pretty showstopping but rn you can only know that if you have eyes on the console. probably want to return the showstopping errors. Not necessarily applicable to loop where one can fail and you want to keep going but the intiializing stuff should proibably return errors

Reply:
might want to go for some nicer logging than fmt.Println tho